### PR TITLE
Marathi ITN

### PR DESCRIPTION
**Pull request by ChinmayPatil11**
# What does this PR do ?
This is a PR for marathi ITN support for Cardinal, Decimal, Date and Time categories.

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Have you signed your commits? Use ``git commit -s`` to sign.
- [x] Do all unittests finish successfully before sending PR?
   1) ``pytest`` or (if your machine does not have GPU) ``pytest --cpu`` from the root folder (given you marked your test cases accordingly `@pytest.mark.run_only_on('CPU')`).
   2) Sparrowhawk tests ``bash tools/text_processing_deployment/export_grammars.sh --MODE=test ...``
- [x] If you are adding a new feature: Have you added test cases for both `pytest` and Sparrowhawk [here](tests/nemo_text_processing).
- [x] Have you added ``__init__.py`` for every folder and subfolder, including `data` folder which has .TSV files?
- [x] Have you followed codeQL results and removed unused variables and imports (report is at the bottom of the PR in github review box) ?
- [x] Have you added the correct license header `Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.` to all newly added Python files?
- [x] If you copied [nemo_text_processing/text_normalization/en/graph_utils.py](nemo_text_processing/text_normalization/en/graph_utils.py) your header's second line should be `Copyright 2015 and onwards Google, Inc.`. See an example [here](https://github.com/NVIDIA/NeMo-text-processing/blob/main/nemo_text_processing/text_normalization/en/graph_utils.py#L2).
- [x] Remove import guards (`try import: ... except: ...`) if not already done.
- [x] If you added a new language or a new feature please update the [NeMo documentation](https://github.com/NVIDIA/NeMo/blob/main/docs/source/nlp/text_normalization/wfst/wfst_text_normalization.rst) (lives in different repo).
- [x] Have you added your language support to [tools/text_processing_deployment/pynini_export.py](tools/text_processing_deployment/pynini_export.py).
  

**PR Type**:
- [x] New Feature
- [x] Documentation

@ekmb

### CODE REVIEW

**Comment by ChinmayPatil11** [comment]
Hi @ekmb Can you please check and approve on the PR? I have added a PR on the NeMo repo as well with added feature. Thanks.

File: nemo_text_processing/inverse_text_normalization/mr/data/date/dates.tsv [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+१७	सतरा
+१८	अठरा
+१९	एकोणीस
+२०	वीस
+२१	एकवीस
+२२	बावीस
+२३	तेवीस
+२४	चोवीस
+२५	पंचवीस
+२६	सहावीस
```
Comment (virajkarandikar) [inline nemo_text_processing/inverse_text_normalization/mr/data/date/dates.tsv:26]:
```suggestion
२६	सव्वीस 
```

File: nemo_text_processing/inverse_text_normalization/mr/data/numbers/tens.tsv [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+७६	शहात्तर
+७७	सत्याहत्तर
+७८	अठ्ठ्यात्तर
+७९	एकोणऐंशी
+८०	ऐंशी
+८१	एक्याऐंशी
+८२	ब्याऐंशी
+८३	त्र्याऐंशी
+८४	चौऱ्याऐंशी
+८५	पंच्याऐशी
```
Comment (virajkarandikar) [inline nemo_text_processing/inverse_text_normalization/mr/data/numbers/tens.tsv:76]:
```suggestion
८५	पंच्याऐंशी
```

File: nemo_text_processing/inverse_text_normalization/mr/data/time/minutes.tsv [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+पस्तीस	३५
+छत्तीस	३६
+सदतीस	३७
+अडतीस	३८
+एकोणचाळीस	३९
+चाळीस	४०
+एकेचाळीस	४१
+बेचाळीस	४२
+त्रेचाळीस	४३
+चौरेचाळीस	४४
```
Comment (virajkarandikar) [inline nemo_text_processing/inverse_text_normalization/mr/data/time/minutes.tsv:44]:
```suggestion
चव्वेचाळीस ४४
```

File: nemo_text_processing/inverse_text_normalization/mr/data/time/minutes.tsv [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+पंचेचाळीस	४५
+सेहचाळीस	४६
```
Comment (virajkarandikar) [inline nemo_text_processing/inverse_text_normalization/mr/data/time/minutes.tsv:46]:
```suggestion
सेहेचाळीस ४६
```

File: nemo_text_processing/inverse_text_normalization/mr/data/time/minutes_to.tsv [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+पस्तीस	२५
+छत्तीस	२४
+सदतीस	२३
+अडतीस	२२
+एकोणचाळीस	२१
+चाळीस	२०
+एकेचाळीस	१९
+बेचाळीस	१८
+त्रेचाळीस	१७
+चौरेचाळीस	१६
```
Comment (virajkarandikar) [inline nemo_text_processing/inverse_text_normalization/mr/data/time/minutes_to.tsv:44]:
```suggestion
चव्वेचाळीस	१६
```

**Comment by ekmb** [comment]
@ChinmayPatil11 thank you for the PR!

Could you please address CodeQL errors.

**Comment by ChinmayPatil11** [comment]
> @ChinmayPatil11 thank you for the PR!
> 
> Could you please address CodeQL errors.

Sure I will, thanks for the review. Meanwhile can you please remove the stale tag on the corresponding PR to NeMo repository? https://github.com/NVIDIA/NeMo/pull/8117
Thanks!

File: nemo_text_processing/inverse_text_normalization/mr/data/time/hours_to.tsv [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+तीन	२
+चार	३
+पाच	४
+सहा	५
+सात	६
+आठ	७
+नऊ	८
+दहा	९
+अकरा	१०
+बारा	११
```
[the remark this answers is not in the document]
Comment (ChinmayPatil11) [reply nemo_text_processing/inverse_text_normalization/mr/data/time/hours_to.tsv:12]: Yes, these are for case of 'पावणे <hour>' in Marathi. Similar to 'Quarter to <hour>' case in English

File: nemo_text_processing/inverse_text_normalization/mr/data/date/dates.tsv (hunk 1 above)
Comment (ChinmayPatil11) [reply nemo_text_processing/inverse_text_normalization/mr/data/date/dates.tsv]: Thank you for the review. Have changed all the files with the errors.

**Comment by ChinmayPatil11** [comment]
Hi @ekmb, can you please check on the review. I have done the required changes on my part. Thanks!

**Comment by ekmb** [comment]
> Hi @ekmb, can you please check on the review. I have done the required changes on my part. Thanks!

Hi @ChinmayPatil11 , could you please add pytests + sparrowhawk tests to the jenkins file? Examples: 
https://github.com/NVIDIA/NeMo-text-processing/pull/134/files# [create cache files]
https://github.com/NVIDIA/NeMo-text-processing/blob/main/Jenkinsfile#L333 [run all pytests]
https://github.com/NVIDIA/NeMo-text-processing/blob/main/Jenkinsfile#L393 [run SH tests]

Thanks!

**Comment by tbartley94** [comment]
jenkins

**Comment by tbartley94** [comment]
jenkins

**Comment by tbartley94** [comment]
jenkins

**Review by ekmb — APPROVED** [review]

**Pull request merged**